### ISSUE TITLE
Snippets discussion - bracing style

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/vscode-powershell#62]
I (reluctantly) would like to open a discussion regarding which bracing style to use for snippets.  Right now, the latest snippets file uses Allman style bracing e.g.:
```powershell
function foo
{
    ...
}
```
But the bulk of the PowerShell code I've seen in the wild uses K&R style e.g.:
```powershell
function foo {
    ...
}
```
I know there are details of bracing style the community at large doesn't agree upon like whether or not to use cuddled else but I do believe K&R is fairly well accepted at the "top level".

Would anyone object strenuously if I changed the snippets to use K&R?

**Comment by daviwil** [comment]
I have no objection to that since it seems to be the prevalent coding style in PowerShell.  I'm not sure who wrote the original snippets but we should definitely make them conform to generally expected PowerShell coding style.

**Comment by rkeithhill** [comment]
OK we are going with K&R style which I see more often out in the wild in PowerShell scripts.

**Comment by Jaykul** [comment]
I vote K&R 
With cuddled else!
Because ... code folding.

Oh, right. VS Code doesn't do that yet. :-P

**Comment by rkeithhill** [comment]
With the current snippets for `if`, `elseif` and `else`, you can control whether you get cuddling or not by where you invoke the snippet.  The following will give you a cuddled else:
```powershell
if ($foo) {
    ...
} else{tab}
```
And if you want standard K&R, invoke the snippet below the `if` statement's closing brace e.g.:
```powershell
if ($foo) {
    ...
} 
else{tab}
```

**Comment by rkeithhill** [comment]
This has been implemented for the upcoming 0.4.0 release.

**Comment by thegraffix** [comment]
In the future, would it be possible to introduce 2x different PowerShell.json files for each bracing style (K&R and Allman) so the user can choose an appropriate style?  Some organizations don't enforce a bracing style standard for PowerShell so it would be really useful to be able to switch between the two on the fly without the need to manually edit the snippets in the PowerShell.json file.
